src/http_server/api/v1/health.c

```diff
@@ -314,9 +489,76 @@ static void cb_health(mk_request_t *request, void *data)
     }
 }
 
+static void configure_throughput_check(struct flb_config *config)
+{
+    bool enabled = config->hc_throughput;
+
+    throughput_check_state.enabled = false;
+    throughput_check_state.healthy = true;
+
+    if (!enabled) {
+        return;
+    }
+
+    if (!config->hc_throughput_input_plugins) {
+        flb_warn("[api/v1/health/throughput]: " FLB_CONF_STR_HC_THROUGHPUT_IN_PLUGINS " is required");
+        return;
+    }
+    if (!config->hc_throughput_output_plugins) {
+        flb_warn("[api/v1/health/throughput]: " FLB_CONF_STR_HC_THROUGHPUT_OUT_PLUGINS " is required");
+        return;
+    }
+    if (!config->hc_throughput_ratio_threshold) {
+        flb_warn("[api/v1/health/throughput]: " FLB_CONF_STR_HC_THROUGHPUT_RATIO_THRESHOLD " is required");
+        return;
+    }
+    if (!config->hc_throughput_min_failures) {
+        flb_warn("[api/v1/health/throughput]: " FLB_CONF_STR_HC_THROUGHPUT_MIN_FAILURES " is required");
+        return;
+    }
+
+    throughput_check_state.sample_list = flb_malloc(sizeof(struct mk_list));
+    if (!throughput_check_state.sample_list) {
+        flb_errno();
+        return;
+    }
+    mk_list_init(throughput_check_state.sample_list);
+
+    throughput_check_state.input_plugins =
+        flb_utils_split(config->hc_throughput_input_plugins, ',', 0);
+
+    if (!throughput_check_state.input_plugins) {
+        flb_errno();
+        return;
```

Comment:
You need to release the memory allocated in line 520 in this case (same goes for 540)